Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2184,7 +2184,7 @@ void WebPageProxy::updateActivityState(OptionSet<ActivityState::Flag> flagsToUpd
         m_activityState.add(ActivityState::IsAudible);
     if (flagsToUpdate & ActivityState::IsLoading && m_pageLoadState.isLoading())
         m_activityState.add(ActivityState::IsLoading);
-    if (flagsToUpdate & ActivityState::IsCapturingMedia && m_mediaState.containsAny({ MediaProducerMediaState::HasActiveAudioCaptureDevice,  MediaProducerMediaState::HasActiveVideoCaptureDevice }))
+    if (flagsToUpdate & ActivityState::IsCapturingMedia && m_mediaState.containsAny(MediaProducer::ActiveCaptureMask))
```

Comment:
Why isn't this change sufficient? It should communicate to the WebPage that it cannot be throttled and I would expect WebPage::m_activityState to prevent AppNap even screen recording now. As a result, it is not clear to me why we need an extra/explicit UserActivity in WebCore (which is the first one of its kind).